### PR TITLE
fix: improve tokern error message inside non ctx cmd

### LINKED ISSUE #2868: okteto deploy inside container ask you to add `--token` flag but it can't be added
**Describe the bug**
If you run okteto up and inside the container you run an okteto deploy command. The command will fail with the following error:
Captura de pantalla 2022-06-30 a las 21 45 30

But when adding the flag to the command it will fail with  `x  Unknown flag: --token`

**To Reproduce**

1. Copy the following okteto manifest:
```yaml
name: aaaa
image: okteto/golang-ci:1.17.10
command: bash
sync:
- .:/usr/src/app
autocreate: true
```
2. Run `okteto up` 
3. Run `curl https://get.okteto.com -sSfL | sh` inside the container
4. Create the following okteto manifest inside the container:
```yaml
deploy:
  ls
```
5. Run `okteto deploy` from the path of the new okteto manifest
6. See the image error
7.  Run `okteto deploy --token YOUR_TOKEN` and check that it returns `x  Unknown flag: --token`

**Expected behavior**
Be able to run the command or return a proper error without mentioning `--token` flag

**Desktop (please complete the following information):**

- Version 2.4.1

### PR DESCRIPTION
**Pull request by jLopezbarb**
Signed-off-by: Javier López Barba <javier@okteto.com>

# Proposed changes

Fixes #2868

- Update error message when load context is inside the cluster but is a non ctx command

Captura de Pantalla 2022-09-06 a las 16 03 37

### CODE REVIEW

**Review by AdrianPedriza — APPROVED** [review]

File: pkg/errors/errors.go [hunk 1]
```diff
@@ -107,9 +107,12 @@ var (
 	// ContextIsNotOktetoCluster raised if the cluster connected is not managed by okteto
 	ErrContextIsNotOktetoCluster = fmt.Errorf("this command is only available on Okteto Cloud or Okteto Enterprise")
 
-	// ErrTokenFlagNeeded is raised when the command is executed from inside a pod
+	// ErrTokenFlagNeeded is raised when the command is executed from inside a pod from a ctx command
 	ErrTokenFlagNeeded = fmt.Errorf("this command is not supported without the '--token' flag from inside a container")
 
+	// ErrTokenEnvVarNeeded is raised when the command is executed from inside a pod from a non ctx command
+	ErrTokenEnvVarNeeded = fmt.Errorf("this command is not supported without the 'OKTETO_TOKEN' environment variable set from inside a container")
```
Comment (rberrelleza) [inline pkg/errors/errors.go:114]: "The OKTETO_TOKEN environment variable is required when running this command from within a container."

Comment (rberrelleza) [reply pkg/errors/errors.go:114]: @jmacelroy does that error message sounds alright?

**Review by teresaromero — APPROVED** [review]
i think you should confirm the copy for the error, but other than that LGTM

File: pkg/errors/errors.go [hunk 2]
```diff
@@ -107,9 +107,12 @@ var (
 	// ContextIsNotOktetoCluster raised if the cluster connected is not managed by okteto
 	ErrContextIsNotOktetoCluster = fmt.Errorf("this command is only available on Okteto Cloud or Okteto Enterprise")
 
-	// ErrTokenFlagNeeded is raised when the command is executed from inside a pod
+	// ErrTokenFlagNeeded is raised when the command is executed from inside a pod from a ctx command
 	ErrTokenFlagNeeded = fmt.Errorf("this command is not supported without the '--token' flag from inside a container")
 
+	// ErrTokenEnvVarNeeded is raised when the command is executed from inside a pod from a non ctx command
+	ErrTokenEnvVarNeeded = fmt.Errorf("the 'OKTETO_TOKEN' environment variable is required when running this command from within a container")
```
Comment (AgustinRamiroDiaz) [inline pkg/errors/errors.go:114]: Can we add a link to the docs on how to retrieve the `OKTETO_TOKEN`?

**Review by AgustinRamiroDiaz — APPROVED** [review]

**Pull request merged**